tldw_Server_API/tests/Explainer/test_explainer_chatbook_export.py

```
@@ -0,0 +1,437 @@
+
+def test_restore_rejects_payload_with_no_dict_nodes(tmp_path):
+    target_repo = ExplainerRepository(ExplainerDatabase(tmp_path / "target.db"))
+    payload = {
+        "type": "explainer_session",
+        "structured": {
+            "session": {"title": "Broken"},
+            "nodes": ["not-a-node", 42],
+        },
+    }
+
+    with pytest.raises(ValueError):
```

Ruff (0.15.15)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)